### DISCUSSION TITLE
Building business on reversed engineered api

### DISCUSSION THREAD

**Discussion by aleifsm** [timdorr/tesla-api#481] [General]
What are your thoughts on building a business that relies on this API, since I need my clients to be able to unlock Tesla cars and remotely start them using this API. 
And what does Tesla think about this project, do you think they will closed all 3rd party API calls in the future?

**Comment by JonG67x** [comment]
Generating the basic tokens to use the API has stopped working again for many in the last couple of days, It seems to be a cat and mouse game with Tesla and the fact this year they've tried so hard to stop it makes you wonder whether they will continue to tighten the ability to do things until they prevent third party access. If you have the tokens then things are fairly stable, its getting the tokens at the moment that is proving the challenge. I personally wouldn't want to either enter into a contract with a third party reliant on the Tesla API at the moment.
